internal/edgeapi/backend/k8s/backend.go

```diff
@@ -46,20 +42,17 @@ func NewBackend(repository RepositoryFacade, assembler *ConfigurationAssembler,
 func (b *backend) GetRegistrationStatus(ctx context.Context, name, namespace string) (backendapi.RegistrationStatus, error) {
 	edgeDevice, err := b.repository.GetEdgeDevice(ctx, name, namespace)
 	if err != nil {
+		if errors.IsNotFound(err) {
+			return backendapi.Unregistered, nil
```

Comment:
Could it be that the device missing, but the EDSR was created?
So the actual status is pending approval?

Reply:
It is possible that there is EDSR and not EdgeDevice, but `GetRegistrationStatus` would never be called in that situation - `control` requests are handled only after device is successfully registered and uses proper, device-specific, non-registration certificate.